File: leetcode-121.py
from typing import List

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minx = int(1e9)
        ans = 0
        for v in prices:
            ans = max(ans, v-minx)
            minx = min(v, minx)
        return ans


if __name__ == "__main__":
    so = Solution()
    print(so.maxProfit([7,1,5,3,6,4]))

File: leetcode/leetcode-121.cpp
/*
 * @File Name          :leetcode-121.cpp
 * @Author             :cc
 * @Version            :1.0.0
 * @Date               :2022/1/22 10:14
 * @Description        :
 * @Function List      :
 * @History            :
 */

#include "vector"
#include "algorithm"

using namespace std;

class Solution {
public:
    int maxProfit(vector<int> &prices) {
        int ans = 0;
        int pre = prices[0];
        for (int i = 1; i < prices.size(); i++) {
            ans = max(ans, prices[i] - pre);
            pre = min(pre, prices[i]);
        }
        return ans;
    }
};